rs/consensus/src/consensus/validator.rs

```diff
@@ -977,6 +1053,35 @@ impl Validator {
                 }
             }
 
+            // Disqualify rank if equivocation was found. If there already
+            // exists a validated block of the same rank as the current
+            // proposal, we must generate an equivocation proof.
```

Comment:
> we must generate an equivocation proof

Currently we don't generate equivocation proofs for proposals that have a higher rank than the lowest `valid_qualified_rank` (they are skipped in the block above). Is that intentional?

Reply:
Yes, that was intentional. Originally I had this check below `is_time_to_make_block` as well. My thinking was that I didn't want a malicious blockmaker to be able to cause unnecessary work at almost every round (so eager vs. lazy).

In practice, this shouldn't make a big difference, so for simplicity's sake I can put the proof generation before the block above. What's your take on this?

Reply:
I see that makes sense. Now that I look at it more, I also don't think it matters if the check is above or below `is_time_to_make_block` either. In order to create an equivocation proof, one of the proposals of that rank needs to be validated, which can only happen after `is_time_to_make_block` was true at least once.

One thing I would expect though is that the check if a rank is disqualified and the routine doing the disqualification would be close to each other. So maybe we can move both of them below `is_time_to_make_block`?

Reply:
Ok, I moved both of them below the time_to_make_block check.